### ISSUE TITLE
After purge and reinstall microk8s, the ingress is never working until a reboot.

### ISSUE THREAD

**Issue by rockkoca** [canonical/microk8s#1911]
sudo snap remove --purge microk8s
sudo snap install microk8s --classic

After using the above command to purge and reinstall microk8s, the ingress for this node is never working after joining a cluster until a reboot. I tried to stop and start microk8s and microk8s.reset as well. Only a reboot can bring the ingress online.

I recently recreated clusters many times. So, I have tried this procedure on different machines and different clusters. I can reproduce this every time. 

I am using Ubuntu 20.04.1 LTS x86_64 and v1.20.1 for microk8s.

**Comment by bartecargo** [comment]
@rockkoca thanks! It won't get back the hours I've already wasted trying to figure out what I was doing wrong, but you've saved me hours by reporting this issue and the workaround.

**Comment by nzbart** [comment]
@rockkoca thanks! It won't get back the hours I've already wasted trying to figure out what I was doing wrong, but you've saved me even more hours by reporting this issue and the workaround.

**Comment by luborpetr** [comment]
Hello,

I've experienced exactly the same behavior today.
I accidentally run `sudo snap remove --purge microk8s` on one of my lab servers, where I had ingress enabled.

So, I did proceed with reinstallation by running `sudo snap install microk8s --classic --channel=1.21`.  
Important to notice, I was installing exactly the same version of the microk8s I had before (microk8s  v1.21.7   2694   1.21/stable).

After reinstall, the ingress was no more functional.
So, I tried to disable and enable the ingress, run `iptables -F` and then `iptables -X` and restart the microk8s.

None of above helped. At the end I had to reboot the machine and it resolved the problem.

My host is Ubuntu 20.04.3 LTS.
